python/opae.admin/opae/admin/tools/fpgareg.py

```diff
@@ -0,0 +1,367 @@
+#! /usr/bin/env python3
+# Copyright(c) 2021, Intel Corporation
+#
+# Redistribution  and  use  in source  and  binary  forms,  with  or  without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of  source code  must retain the  above copyright notice,
+#  this list of conditions and the following disclaimer.
+# * Redistributions in binary form must reproduce the above copyright notice,
+#  this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+# * Neither the name  of Intel Corporation  nor the names of its contributors
+#   may be used to  endorse or promote  products derived  from this  software
+#   without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING,  BUT NOT LIMITED TO,  THE
+# IMPLIED WARRANTIES OF  MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED.  IN NO EVENT  SHALL THE COPYRIGHT OWNER  OR CONTRIBUTORS BE
+# LIABLE  FOR  ANY  DIRECT,  INDIRECT,  INCIDENTAL,  SPECIAL,  EXEMPLARY,  OR
+# CONSEQUENTIAL  DAMAGES  (INCLUDING,  BUT  NOT LIMITED  TO,  PROCUREMENT  OF
+# SUBSTITUTE GOODS OR SERVICES;  LOSS OF USE,  DATA, OR PROFITS;  OR BUSINESS
+# INTERRUPTION)  HOWEVER CAUSED  AND ON ANY THEORY  OF LIABILITY,  WHETHER IN
+# CONTRACT,  STRICT LIABILITY,  OR TORT  (INCLUDING NEGLIGENCE  OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,  EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+import re
+import os
+import argparse
+import struct
+import mmap
+import mmap
+import sys
+import binascii
+import glob
+import stat
+import time
+from ctypes import c_uint64, Structure, Union, c_uint32
+
+
+MAPSIZE = mmap.PAGESIZE
+MAPMASK = MAPSIZE-1
+
+FPGA_OFS_DEVID = ["0xbcce", "0xbccf", "0xaf00"]
+
+PATTERN = (r'.*(?P<segment>\w{4}):(?P<bus>\w{2}):'
+           r'(?P<dev>\w{2})\.(?P<func>\d).*')
+
+BDF_PATTERN = re.compile(PATTERN, re.IGNORECASE)
+
+HSSI_FEATURE_ID = 0x15
+PCIE_DFH_CSR_LEN = 30
+HSSI_DFH_CSR_LEN = 43
+ACCELERATOR_CSR_LEN = 30
+
+
+def verify_pcie_address(pcie_address):
+    """checks format of PCIe address string of the form bb:dd.f or
+    ssss:bb:dd.f, Returns true if  PCIe address of the form ssss:bb:dd.f,
+    or False if addr is not formatted correctly.
+    """
+    m = BDF_PATTERN.match(pcie_address)
+    if m is None:
+        print("Invalid pcie address foramt", pcie_address)
```

Comment:
Two issues:

1. s/foramt/format/
2. There is no format specifier {} in the string to output pcie_address.